.github/workflows/ci_cd.yml

```diff
@@ -923,7 +926,10 @@ jobs:
   deploy-api:
     name: Deploy staging API
     runs-on: ubuntu-latest
-    if: github.event_name == 'push' && (needs.get-changes.outputs.ingestion_server == 'true' || needs.get-changes.outputs.api == 'true')
+    if: |
+      !failure() && !cancelled() &&
+      github.event_name == 'push' && needs.get-changes.outputs.api == 'true' &&
+      (needs.test-ing.result == 'success' && needs.publish-images.result == 'success')
```

Comment:
A similar change could be needed for the `deploy-frontend` action above which also depends on the sometimes-skipped `nuxt-playwright-e2e`, `nuxt-playwright-vr` and `storybook-playwright` jobs.

Reply:
Ah, thanks for that callout!

Reply:
I've taken a stab at this here: 8bb3fef7823b7512ce11690b512cdd790f8cae31 - do you think that looks right?

Reply:
That looks logically right 👍. You could break it into several lines for readability but nonetheless, it should work just fine.